### PR TITLE
Configurable max lead distance

### PR DESCRIPTION
**Pull request by Machine-Maker**

### CODE REVIEW

File: Spigot-Server-Patches/0643-Configurable-max-lead-distance.patch [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index 80409c4b52a4bb7146760070dae0e04d49bdd6b3..6bf99b80cd3351b3c61303967aac4a681f06db1c 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -157,6 +157,12 @@ public class PaperWorldConfig {
+         }
+     }
+ 
++    public float maxLeadDistance = 10f;
```
Comment (kashike) [inline Spigot-Server-Patches/0643-Configurable-max-lead-distance.patch]:
```suggestion
+    public float maxLeashDistance = 10f;
```
It's called a leash elsewhere - your log message, event names, method names.

File: Spigot-Server-Patches/0643-Configurable-max-lead-distance.patch [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
++    private void maxLeadDistance() {
++        maxLeadDistance = getFloat("max-lead-distance", maxLeadDistance);
```
Comment (kashike) [inline Spigot-Server-Patches/0643-Configurable-max-lead-distance.patch]:
```suggestion
+        maxLeashDistance = getFloat("max-leash-distance", maxLeashDistance);
```

File: Spigot-Server-Patches/0643-Configurable-max-lead-distance.patch [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index 80409c4b52a4bb7146760070dae0e04d49bdd6b3..6bf99b80cd3351b3c61303967aac4a681f06db1c 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -157,6 +157,12 @@ public class PaperWorldConfig {
+         }
+     }
+ 
++    public float maxLeadDistance = 10f;
++    private void maxLeadDistance() {
```
Comment (kashike) [inline Spigot-Server-Patches/0643-Configurable-max-lead-distance.patch]:
```suggestion
+    private void maxLeashDistance() {
```

File: Spigot-Server-Patches/0643-Configurable-max-lead-distance.patch [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+diff --git a/src/main/java/net/minecraft/server/EntityCreature.java b/src/main/java/net/minecraft/server/EntityCreature.java
+index 4757e5a92c9cf04fa52d62bac6dae782e08ddab1..965357602df8e1070bdc2178f4a0399b0b0b7ef2 100644
+--- a/src/main/java/net/minecraft/server/EntityCreature.java
++++ b/src/main/java/net/minecraft/server/EntityCreature.java
+@@ -40,7 +40,7 @@ public abstract class EntityCreature extends EntityInsentient {
+             float f = this.g(entity);
+ 
+             if (this instanceof EntityTameableAnimal && ((EntityTameableAnimal) this).isSitting()) {
+-                if (f > 10.0F) {
++                if (f > entity.world.paperConfig.maxLeadDistance) { // Paper
```
Comment (kashike) [inline Spigot-Server-Patches/0643-Configurable-max-lead-distance.patch]:
```suggestion
+                if (f > entity.world.paperConfig.maxLeashDistance) { // Paper
```

File: Spigot-Server-Patches/0643-Configurable-max-lead-distance.patch [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                     this.world.getServer().getPluginManager().callEvent(new EntityUnleashEvent(this.getBukkitEntity(), EntityUnleashEvent.UnleashReason.DISTANCE)); // CraftBukkit
+                     this.unleash(true, true);
+                 }
+@@ -49,7 +49,7 @@ public abstract class EntityCreature extends EntityInsentient {
+             }
+ 
+             this.x(f);
+-            if (f > 10.0F) {
++            if (f > entity.world.paperConfig.maxLeadDistance) { // Paper
```
Comment (kashike) [inline Spigot-Server-Patches/0643-Configurable-max-lead-distance.patch]:
```suggestion
+            if (f > entity.world.paperConfig.maxLeashDistance) { // Paper
```

File: Spigot-Server-Patches/0643-Configurable-max-lead-distance.patch [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -157,6 +157,12 @@ public class PaperWorldConfig {
+         }
+     }
+ 
++    public float maxLeadDistance = 10f;
++    private void maxLeadDistance() {
++        maxLeadDistance = getFloat("max-lead-distance", maxLeadDistance);
++        log("Max leash distance: " + maxLeadDistance);
```
Comment (kashike) [inline Spigot-Server-Patches/0643-Configurable-max-lead-distance.patch]:
```suggestion
+        log("Max leash distance: " + maxLeashDistance);
```

**Review by kashike — CHANGES_REQUESTED** [review]

**Review by MiniDigger — APPROVED** [review]
lgtm

**Review by kashike — APPROVED** [review]

**Pull request merged**